File: frontend/src/app/shared/responses/home-images.response.ts
export class HomeImagesResponse {

  images: HomeImage[];

  constructor(images: HomeImage[]) {
    this.images = images;
  }

}

export class HomeImage {

  id: number;
  name: string;
  content: string;

  constructor(id: number, name: string, content: string) {
    this.id = id;
    this.name = name;
    this.content = content;
  }

}

export class HomeImageWithBytes {
  id: number;
  name: string;
  content: string;
  bytes: string;

  constructor(id: number, name: string, content: string, bytes: string) {
    this.id = id;
    this.name = name;
    this.content = content;
    this.bytes = bytes;
  }
}

File: src/main/java/com/sgorokh/TimeSaver/controllers/responses/HomeImagesResponse.java
package com.sgorokh.TimeSaver.controllers.responses;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class HomeImagesResponse {

    private List<HomeImage> images = new ArrayList<>();

    public void addImage(Long id, String name, String content) {
        images.add(new HomeImage(id, name, content));
    }

    @Data
    @AllArgsConstructor
    public static class HomeImage {

        private Long id;
        private String name;
        private String content;

    }
}
